### ISSUE TITLE
Optional preemptive task scheduler

### ISSUE THREAD

**Issue by brais-sg** [blocksds/sdk#80]
Hi,
right now there's a cooperative task system in Blocks DS that works really well if you call yield accordingly. 
Will be possible to implement an optional preemtive task scheduler, that when enabled can use a timer triggered ISR to switch context from one task to the next one (for example)?

**Comment by AntonioND** [comment]
I agree this would be nice, but it's a lot more complex than it looks. A lot of functions of libnds would have to be made thread safe. I'll leave the issue open, though.

**Comment by AntonioND** [comment]
https://codeberg.org/blocksds/sdk/issues/80
